src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallRepository.kt

```diff
@@ -12,13 +12,16 @@ import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.managerecallsapi.domain.RecallId
 import uk.gov.justice.digital.hmpps.managerecallsapi.service.RecallNotFoundException
 import java.util.UUID
+import javax.transaction.Transactional
 
 @Repository("jpaRecallRepository")
 interface JpaRecallRepository : JpaRepository<Recall, UUID>
 
 @NoRepositoryBean
 interface ExtendedRecallRepository : JpaRecallRepository {
   fun getByRecallId(recallId: RecallId): Recall
+  fun findByRecallId(recallId: RecallId): Recall?
```

Comment:
The 2 methods intentionally behave differently.  'getRecallById' throws a RecallNotFoundException if a recall doesn't exist.  'findByRecallId' returns null if it doesn't find one.   It's standard practice to have a get and a find that behave this way.  e.g. the JpaRecallRepository interface has a getById(UUID) and findById(UUID) that behave the same way.  The methods I have added are just specific to be used with a RecallId instead of UUID

Reply:
fair enough - thanks for clarifying